### ISSUE TITLE
Module : mettis

### ISSUE THREAD

**Issue by gpichot** [rezometz/django-paiji2-mettis#1]
_From @lhark on December 27, 2014 20:12_

As it seems for now, the mettis schedule is retrieved through the UI of http://lemet.fr
but this method is a bit ugly to say the least.

Luckily, Metz has an open data policy and is making quite a lot a data available
through the http://data.gouv.fr portal. In particular, they are providing all of the transport
network data under the GTFS format, which is a standardized format developed by Google
for this very purpose. [GTFS reference](https://developers.google.com/transit/gtfs/reference)
In our case, the .zip can be found [here](https://www.data.gouv.fr/fr/datasets/transport-donnees-gtfs/)

Google is even providing a python library that allow for an easy retrieving of the data :
https://github.com/google/transitfeed

The only problem with this method is that it seems it will be hard to automate the retrieving
of the GTFS file (it is apparently renewed every semester) as data.gouv.fr does not provide
an API for this kind of file.

_Copied from original issue: rezometz/paiji2#33_

**Comment by gpichot** [comment]
Yep Actually if you reverse engineering the Mettis application available on the store you will see that they are using Phonegap. So the website is a single page application developed with AngularJS (which explains performances). This application making call to some API and here follow the urls that you can look at (JSON format):
- http://lemet.fr/apps/fiches.php?action=getStops
- http://lemet.fr/apps/fiches.php?action=getLignes&stop='+$scope.horaires.stop
- http://lemet.fr/apps/fiches.php?action=getHeadsigns&ligne='+$scope.horaires.ligne+'&stop='+$scope.horaires.stop
- http://lemet.fr/apps/horaires.php'+$scope.horaires.headsign
- http://lemet.fr/apps/fiches.php?action=getStopsByLatLng&lat='+$scope.coords.lat+'&lng='+$scope.coords.long
- http://lemet.fr/apps/fiches.php?action=getCompleteStop&stop='+index
- http://lemet.fr/root/Datas/geojson/'+name+'.json'
- http://lemet.fr/apps/fiches.php?action=getStopsByAddress&adr='+adresse
- http://lemet.fr/apps/adresse.php?action=rechercheadresse&adresse='+$adresse
- http://lemet.fr/apps/fiches.php?action=getSimpleLignes
- http://lemet.fr/apps/rss.php?action=getRSS&type=alertes
- http://lemet.fr/apps/rss.php?action=getRSS&type=actus
- http://lemet.fr/apps/adresse.php?action=get_location&lat=' + $scope.coords.lat+"&lng="+$scope.coords.long
  *http://lemet.fr/apps/adresse.php?action=get_address&address=' + adr
- http://lemet.fr/apps/test.php?arret_depart="+$scope.master.dep+"&arret_arrivee="+$scope.master.arr+"&jour="+$scope.master.jour+"&heure_depart="+$scope.master.heure;
- http://lemet.fr/apps/horaires.php'+$scope.selected.remote.direction

I do not think that it has evolved a lot since I fetched these urls. I do not really like how the data is formatted and the API is structured but it is the better choice to go with (I asked them for an API in September '14 and they answered me that they were not providing any API at all).

For implementation, I think we should provide a python API (not in the paiji2 repository) that will work as an interface between the paiji2 module and the above API. I can explain how it works if needed as I already dived into (I think I even have a python API for that, will have to look at it).

**Comment by gpichot** [comment]
_From @lhark on December 28, 2014 14:54_

I just had a look at the API you found : basicaly, they are using the same GTFS data to answer the queries (for example the stop IDs are the sameas the ones i got).
The only thing different is the possibility to get alerts via RSS feed.

We need to choose between a little bit less processing on our side, or a better durability of our module. Indeed the fact that this API is hidden means they one day could chose to completly rewrite it without maintaining backward compatibility, whereas the GTFS format and the transitfeed library are widely used.

**Comment by lhark** [comment]
That's exactly why i believe the GTFS data seems to be a better idea. However, the data seems to be manualy uploaded to data.gouv.fr, let's hope the don't forget about it.
